pkg/reconciler/instances/busola/action.go

```diff
@@ -0,0 +1,112 @@
+package busola
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"k8s.io/client-go/rest"
+	"net/url"
+	"strings"
+)
+
+type virtualServicePatch struct {
+	Spec specPatch `json:"spec"`
+}
+
+type specPatch struct {
+	Hosts []string `json:"hosts"`
+}
+
+type VirtualSvcMeta struct {
+	Name      string
+	Namespace string
+}
+
+type VirtualServicePreInstallPatch struct {
+	name            string
+	virtSvcsToPatch []VirtualSvcMeta
+	suffix          string
+	virtSvcClient   VirtSvcClient
+}
+
+var _ service.Action = &VirtualServicePreInstallPatch{}
+
+func NewVirtualServicePreInstallPatch(virtualSvcs []VirtualSvcMeta, suffix string) *VirtualServicePreInstallPatch {
+	client := NewVirtSvcClient()
+	return &VirtualServicePreInstallPatch{"pre-reconciler", virtualSvcs, suffix, client}
+}
+
+func (p *VirtualServicePreInstallPatch) Run(version string, profile string, configuration []reconciler.Configuration, helper *service.ActionContext) error {
+	ctx := context.TODO()
+	logger := helper.Logger
+	clientSet, err := helper.KubeClient.Clientset()
+	if err != nil {
+		return errors.Wrapf(err, "while getting client set from kubeclient")
+	}
+	restClient := clientSet.Discovery().RESTClient()
+
+	logger.Infof("Launching pre install busola migrator job, version: %s ", version)
+	for _, virtSvcToPatch := range p.virtSvcsToPatch {
+		logger.Infof("Patching virtual service: %s in namespace: %s", virtSvcToPatch.Name, virtSvcToPatch.Namespace)
+		if err := p.patchVirtSvc(ctx, restClient, virtSvcToPatch.Name, virtSvcToPatch.Namespace, logger); err != nil {
```

Comment:
What will happen if the virtual service doesn't exist?

Reply:
There will be an error and the step will fail.
I added test case for this.

Reply:
Then it behaves differently than the current helm hook implementation and having it work without dex/console components installed was one of the requirements if I remember correctly.

Reply:
You're right. I am going to fix it